src/justfile.rs

```diff
@@ -28,19 +28,31 @@ impl<'src> Justfile<'src> {
     self.recipes.len()
   }
 
-  pub(crate) fn suggest(&self, name: &str) -> Option<&'src str> {
+  pub(crate) fn suggest(&self, name: &str) -> Option<Suggestion> {
```

Comment:
This is totally fine, but if you use a different name for the parameter, then you can more easily assign to `Suggestion::name`. using the `Foo{field}` shorthand.

So if it's:

```rust
pub(crate) fn suggest(&self, input: &str) -> Option<Suggestion>
```

Then later on you can do:

```rust
recipe
.keys()
.map(|name| {
 (edit_distance(input, name), Suggestion { name, target: None})
})
```

And then later on:

```rust
aliases
.iter()
.map(|name, alias| {
 (edit_distance(input, name), Suggestion { name, target: Some(alias.target.name.lexem())})
})
```